### PR TITLE
Angular v22 support

### LINKED ISSUE #1043: Is support for angular v22 still planned
Hello,

we used this library now already for many many years and I'm wondering if the UI-Router lib will also support in the near future v22 or if this won't be supported anymore? The latest version that ui-router currently supports is v21 (at the moment of writing this msg).

Thanks for any kind of information.

### PR DESCRIPTION
**Pull request by lindolo25**
Hi @wawyed and @christopherthielen!

Here is the implementation to support Angular v22, it also includes adding Angular cli for easier updates in the future. Tests had few changes as well, configuration and test cases due to changes to Angular API. If you have any questions about the changes please let me know. As always thanks for your time and help!

### CODE REVIEW

**Comment by christopherthielen** [comment]
@lindolo25 should we be using an explicit ChangeDetectionStrategy in UIView?  [The default changed from Eager to OnPush](https://github.com/angular/angular/blob/083593e4f55dcbfa7b40371daa182e2b8fe48f70/CHANGELOG.md#L3-L31) in NG22, from what I'm reading.  I also noted that [a previous commit](https://github.com/ui-router/angular/commit/2ca4491b780a4bb0b4718e479a107a5c599b9b2e) last year turned off the OnPush strategy during the NG21 release?

**Comment by lindolo25** [comment]
Hi @christopherthielen 

Thanks for your time reviewing this. Yes I think having the explicit value of change detection set for the component will give clarity for readers of what's the intended value. And yes the default value changed from Eager to OnPush with this new version. I'll add the change myself with a new commit.

Regarding what value to use, in my opinion should be OnPush. I Added this value before withing  [commit a647147](https://github.com/ui-router/angular/commit/a6471473a79a5352e9d30d8bb0bba08759dedf82) and reverted it with [commit 2ca4491](https://github.com/ui-router/angular/commit/2ca4491b780a4bb0b4718e479a107a5c599b9b2e). As you said both of this commits are part of the changes for v21. Why I removed it, was my lack of understanding at the time. I was unsure of what effect this change could have, so I played it save. Since then I've tested and used the OnPush withing the projects I work with and I'm confident the UiView will continue to work as expected. 

* UIView already handles how component's inputs are set, when a component ref is created. There will be no impact here.
* Withing it's template, is using signals for changes that need to be detected. This will be OK.
* And all testing pass.

please let me know your thoughts

**Comment by lindolo25** [comment]
Well not all tests are passing! I need to investigate more.

**Comment by christopherthielen** [comment]
OK, let me know how you want to proceed.  

I'm only aware of the differences between the two, but I do not understand nuances that may effect edge cases.   If it makes sense to ship with `Eager`, that's fine by me.

**Comment by christopherthielen** [comment]
Ran this through a 🤖 robot 🤖 and it thinks the failing tests are due to child components doing direct variable mutations in async updates without a `markForCheck` call:

> *What happens*:
> 1. The routed AppComponent owns a dynamically created dialog. In [DialogComponent.ngOnInit](https://github.com/ui-router/sample-app-angular/blob/e97336d13af6d18964aef41808657af799927abc/src/app/global/dialog.component.ts#L25-L55), a timer/promise waits 50 ms and then does a plain this.visible = true.
> 2. Zone schedules a change-detection pass, but [UIView is now a clean OnPush ancestor](https://github.com/ui-router/angular/blob/ecb5f5f69663902b028cc593edcf76c6694f0bb9/src/directives/uiView.ts#L113-L123). Angular skips that entire routed subtree. The fact that the routed components themselves are Eager/Default does not override a clean OnPush ancestor.
> 3. The dialog’s @HostBinding('class.active') therefore never reaches the DOM. Its CSS leaves the content at top: -200px instead of moving it to top: 4em ([dialog CSS](https://github.com/ui-router/sample-app-angular/blob/e97336d13af6d18964aef41808657af799927abc/src/styles.css#L278-L321)), exactly matching CI’s “button is outside of the viewport.”
>
> I reproduced [the failing test](https://github.com/ui-router/angular/actions/runs/31012556138/job/92328205702) locally. Adding ChangeDetectorRef.markForCheck() immediately after visible = true makes it pass.
>
> *My recommendation*: ship UIView as explicit ChangeDetectionStrategy.Eager for 22. This isn’t merely an internal optimization—the outlet becomes an OnPush boundary around user-owned application trees. 
>
>UIView cannot generically know about every later async mutation below it, so adding another markForCheck() around route creation won’t solve the class of failures. We could support explicit OnPush later, but it should be an intentional breaking behavior with focused tests for timer/Promise updates, manual subscriptions, uiOnParamsChanged, dynamic child components/overlays, nested views, projected content, and both zone/zoneless apps.

Based on my limited understanding, this seems appropriate to me.  Ship with `Eager`.   If it seems useful add an opt-in `OnPush` mode (this can be done later as a point release).

**Comment by lindolo25** [comment]
Hi @christopherthielen! 

Sorry for the delay, I've been busy these last few days. I'm making the change to use `Eager`. I'll investigate the failing test in details later. With `Eager` I'm getting another error with the downstream tests, this time is not a failing test but rather the build. Apparently the build for the Sample app because is using Angular v21 and trying to import the partial build for Angular v22, It does not know how to handle the `ChangeDetection.Eager`. I'm creating an update for the sample app so the tests can properly run in Angular v22 and azure there is no regression.

I'll add an update here when is ready, thanks for the Help.

**Comment by lindolo25** [comment]
Hi @christopherthielen!

linking here the PR for the sample app. Once merged, I'll add the update to Eager here to trigger the tests. I ran it from my side and is working. Hopefully it will work here as well! 

https://github.com/ui-router/sample-app-angular/pull/800

**Comment by christopherthielen** [comment]
@lindolo25 I merged https://github.com/ui-router/sample-app-angular/pull/800 , thanks!

**Comment by lindolo25** [comment]
Hi @christopherthielen, thanks for all the help! All tests are passing, I'm ready to merge. I'm going to review how the release is done, but at least for this one I'll need your help.

File: package.json [hunk 1]
```diff
@@ -48,37 +48,38 @@
...
+    "chokidar": "^5.0.0",
+    "eslint": "^9.28.0",
     "husky": "^9.0.0",
     "jsdom": "^27.4.0",
-    "ng-packagr": "^21.0.0",
+    "ng-packagr": "^22.0.1",
     "prettier": "^3.4.0",
     "pretty-quick": "^4.0.0",
     "rxjs": "~7.8.2",
     "tslib": "^2.8.1",
```
[the remark this answers is not in the document]
Comment (lindolo25) [reply package.json]: Completed!

**Comment by lindolo25** [comment]
Hi @christopherthielen, I reviewed the release task and notice a commit was needed for the change log to reflect the feature change. I added an empty commit with the proper commit message. I also ran the ESLint, and no changes needed. With the branch merged, the only task left is to run the release with publish credentials for NPM.

**Comment by christopherthielen** [comment]
I released 22.0.0

`npm run release -- --help`

**Comment by Ema85** [comment]
Thank you so much @lindolo25 !!
